### PR TITLE
Split content_store_controller.go

### PR DESCRIPTION
**Pull request by vinayvinay**
https://trello.com/c/PbCzGBIB

in response to [feedback](https://github.com/alphagov/publishing-api/pull/22#discussion_r25173958) on: https://github.com/alphagov/publishing-api/pull/22

why?

1. behaviour for publish intent and content item actions have diverged
2. publish intent controller need not know about draft content store
3. allows HandleFunc names to be verbs describing the actions
4. methods that need not be defined on a controller, are now in a
   separate file within the same package: helpers.go

### CODE REVIEW

File: controllers/helpers.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	httpMethod string, path string, data []byte, w http.ResponseWriter) {
+
+	resp, err := contentStoreClient.DoRequest(httpMethod, path, data)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+
+	if w != nil {
+		if err != nil {
+			renderer.JSON(w, http.StatusInternalServerError, err)
```
Comment (heathd) [inline controllers/helpers.go]: do we need to write log output here as well?

File: controllers/helpers.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	if err != nil {
+		renderer.JSON(w, http.StatusInternalServerError, err)
+		return nil, nil
+	}
+
+	var contentStoreRequest *ContentStoreRequest
+	if err := json.Unmarshal(requestBody, &contentStoreRequest); err != nil {
+		switch err.(type) {
+		case *json.SyntaxError:
+			renderer.JSON(w, http.StatusBadRequest, err)
```
Comment (heathd) [inline controllers/helpers.go]: likewise, if we are writing error responses via http, we should also write a log message. Perhaps this happens automatically, I may well have overlooked something?

**Comment by vinayvinay** [comment]
@heathd: i suggest we review and add logging where ever necessary as a separate PR. i've also been thinking about how to log third-party error messages in our JSON logs. it seemed like a little more effort than just adding a log statement.

i understand that correct logging is a given in every PR. however in this case, i've mostly moved existing code to a new place and there's been quite a churn. https://github.com/alphagov/publishing-api/pull/22 as it is, had too many things to be reviewed, so for convenience i've pulled out this sub-PR, idea being we review and merge smaller changes. if you're happy, i'll add a card to Trello for reviewing and adding logging to publishing-api.

**Comment by alext** [comment]
We already have standard request logging (implemented as [middleware](https://github.com/alphagov/publishing-api/blob/master/main.go#L50)).

**Comment by jamiecobbett** [comment]
This is a really good change - makes it clearer and easier to read. It's always a smell when a code file gets really long. :+1:

**Pull request merged**